sygnal/gcmpushkin.py

```diff
@@ -347,16 +347,21 @@ async def _dispatch_notification_unlimited(
         with self.sygnal.tracer.start_span(
             "gcm_dispatch", tags=span_tags, child_of=context.opentracing_span
         ) as span_parent:
+            # TODO: Implement collapse_key to queue only one message per room.
+            failed: List[str] = []
+
             data = GcmPushkin._build_data(n, device)
+
+            # Reject pushkey if default_payload is misconfigured
+            if data is None:
+                failed.append(device.pushkey)
```

Comment:
looks like this will go on to attempt delivery anyway?

Reply:
It sure does look like that-which probably doesn't make sense. I can fix that along with downgrading the error to a warning.